### PR TITLE
Add Protobuf span reporter

### PR DESCRIPTION
**Pull request by jan25**
## Which problem is this PR solving?
- To support zipkin protobuf spans to `/api/v2` endpoint, this PR adds the reporter logic to send test zipkin `PROTO3` format spans to `jaeger-collector`
- Main issue for supporting Zipkin proto spans over HTTP: https://github.com/jaegertracing/jaeger/pull/1695

## Short description of the changes
- Very similar logic to previously implemented JSON, THRIFT reporters
- I had to upgrade dependencies to latest versions to be able to report PROTO format spans

### CODE REVIEW

**Comment by pavolloffay** [comment]
@jan25 thanks for the PR, it looks good. Could you please fix the CI failures? The xdoc failing due to missing proto endpoint is fine, but there are also other failures.

**Comment by jan25** [comment]
@pavolloffay Thanks for the review. I updated the license headers to fix failing CI.

File: docker-compose.yml [hunk 1]
```diff
@@ -48,6 +49,14 @@ services:
         environment:
             - ENCODING=THRIFT
 
+    zipkin-brave-proto:
+        build: .
+        ports:
+            - "8080-8081"
+        environment:
+            - ENCODING=PROTO
+            - JSON_ENCODER=PROTO3
```
Comment (pavolloffay) [inline docker-compose.yml]: What is this variable for?

File: src/main/java/io/jaegertracing/xdock/zipkin/Application.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public String getServiceName() {
 
     @Bean
     public ZipkinTracing tracer() {
-        if (spanBytesEncoder == SpanBytesEncoder.JSON_V2) {
+        if (spanBytesEncoder == SpanBytesEncoder.JSON_V2 || spanBytesEncoder == SpanBytesEncoder.PROTO3) {
```
Comment (jan25) [inline src/main/java/io/jaegertracing/xdock/zipkin/Application.java]: @pavolloffay I had to add a extra condition to route PROTO spans to v2 endpoint, which is why i added `JSON_ENCODER=PROTO3` in docker-compose.yml

Comment (jan25) [reply src/main/java/io/jaegertracing/xdock/zipkin/Application.java]: Actually, If we notice there are two sets of branches in this method. 1. to route to v1 or v2 endpoint 2. to build a Tracing object, Would be nice if its possible to merge these two branches into one.

Comment (pavolloffay) [reply src/main/java/io/jaegertracing/xdock/zipkin/Application.java]: Can't you use v2 endpoint if only `ENCODING=PROTO` ?

Comment (jan25) [reply src/main/java/io/jaegertracing/xdock/zipkin/Application.java]: Yes we could, but after reading bit of zipkin2 lib, it looks like better way is to use `JSON_ENCODER` throughout, we won't need our own custom `enum Encoding`(probably this was introduced in times of zipkin1). Let me create a separate issue to do this refactor.

**Pull request merged**